lib/poller.cpp

```diff
@@ -196,7 +196,7 @@ void poller_t::wait() {
       ERROR("Catched exception at poller: {}", e.what());
     }
   }
-  if (nfds == 0 && !pd->timer_events.empty()) { // This was a timeout
+  if (wait_ms == 0 && !pd->timer_events.empty()) { // This was a timeout
```

Comment:
I think that if for example next timeout is in x ms, and this time passes, then this check will fail, then do another wait and then succeed. With the previous `nfds == 0`, it will succeed on first loop, no need for unnecessary second loop.